From c0d56f139cbe180d8182aa10647d3055103e986c Mon Sep 17 00:00:00 2001
From: Stefan Chulski <stefanc@marvell.com>
Date: Thu, 14 Apr 2016 12:24:20 +0300
Subject: [PATCH 371/538] fix: net: mvpp2x: phy_mode get moved out of
 fixed-link if

- phy_mode get moved out of fixed-link if

Change-Id: Ic961c76f81fd779ad885ac4edeb30fdd892197e3
Reviewed-on: http://vgitil04.il.marvell.com:8080/29036
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
---
 drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c b/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c
index 028d911..2de87c1 100644
--- a/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c
+++ b/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c
@@ -3417,6 +3417,8 @@ static int mv_pp2_init_emac_data(struct mv_pp2x_port *port,
 	port->mac_data.link_irq = irq_of_parse_and_map(emac_node, 0);
 #endif
 
+	phy_mode = of_get_phy_mode(emac_node);
+
 	if (of_phy_is_fixed_link(emac_node)) {
 		port->mac_data.force_link = true;
 		port->mac_data.link = true;
@@ -3429,12 +3431,9 @@ static int mv_pp2_init_emac_data(struct mv_pp2x_port *port,
 	} else {
 		port->mac_data.force_link = false;
 
-		phy_mode = of_get_phy_mode(emac_node);
-
 		pr_debug("gop_mac(%d), phy_mode(%d) (%s)\n", id,  phy_mode,
 			phy_modes(phy_mode));
 
-
 		switch (phy_mode) {
 		case PHY_INTERFACE_MODE_SGMII:
 			speed = 0;
@@ -3465,8 +3464,8 @@ static int mv_pp2_init_emac_data(struct mv_pp2x_port *port,
 			pr_err("%s: incorrect phy-mode\n", __func__);
 			return -1;
 		}
-		port->mac_data.phy_mode = phy_mode;
 	}
+	port->mac_data.phy_mode = phy_mode;
 	pr_debug("gop_mac(%d), phy_speed(%d)\n", id,  port->mac_data.speed);
 
 
-- 
1.9.1

